app/screens/AppNavigator/screens/Dex/PoolSwap/PoolSwapScreen.tsx

```diff
@@ -0,0 +1,209 @@
+import { AddressToken } from '@defichain/whale-api-client/dist/api/address'
+import { PoolPairData } from '@defichain/whale-api-client/dist/api/poolpair'
+import { MaterialIcons } from '@expo/vector-icons'
+import { StackScreenProps } from '@react-navigation/stack'
+import BigNumber from 'bignumber.js'
+import React, { useEffect, useState } from 'react'
+import { Control, Controller, useForm } from 'react-hook-form'
+import { ScrollView, TouchableOpacity, View } from 'react-native'
+import NumberFormat from 'react-number-format'
+import tailwind from 'tailwind-rn'
+import { Text, TextInput } from '../../../../../components'
+import { getTokenIcon } from '../../../../../components/icons/tokens/_index'
+import { PrimaryButton } from '../../../../../components/PrimaryButton'
+import { PrimaryColor, PrimaryColorStyle } from '../../../../../constants/Theme'
+import { useTokensAPI } from '../../../../../hooks/wallet/TokensAPI'
+import { translate } from '../../../../../translations'
+import LoadingScreen from '../../../../LoadingNavigator/LoadingScreen'
+import { DexParamList } from '../DexNavigator'
+
+interface TokenForm {
+  control: Control
+  controlName: string
+  token: AddressToken
+  onMaxPress?: (amount: string) => void
+  title: string
+}
+
+interface SwapSummaryItems {
+  poolpair: PoolPairData
+  tokenA: AddressToken
+  tokenB: AddressToken
+}
+
+type Props = StackScreenProps<DexParamList, 'PoolSwapScreen'>
+
+export function PoolSwapScreen ({ route }: Props): JSX.Element {
+  const poolpair = route.params.poolpair
+  const tokens = useTokensAPI()
```

Comment:
should refactor this soon but no within this PR, as we don't need hooks to use whale client, the logic can be within redux now